### PR TITLE
Add java samples

### PR DESCRIPTION
**Pull request by lucyzhang929**
Will create a separate PR for Java integration tests

### CODE REVIEW

File: samples/samples-java/pom.xml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        <dependency>
+            <groupId>com.microsoft.azure.functions</groupId>
+            <artifactId>azure-functions-java-library</artifactId>
+            <version>${azure.functions.java.library.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.microsoft.azure.functions</groupId>
+            <artifactId>azure-functions-java-library-sql</artifactId>
+            <version>0.1.0</version>
```
Comment (Charles-Gagnon) [inline samples/samples-java/pom.xml:30]: Add comment next to the library POM value for this version that reminds us to update the samples version as well if that is changed

File: samples/samples-java/pom.xml [hunk 2]
```diff
@@ -0,0 +1,122 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.function</groupId>
+    <artifactId>samples-java</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>jar</packaging>
+
+    <name>Azure Java Functions</name>
```
Comment (Charles-Gagnon) [inline samples/samples-java/pom.xml:10]: Maybe this?

```suggestion
    <name>Java Azure Functions with SQL Bindings</name>
```

File: samples/samples-java/src/main/java/com/function/AddProduct.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            @HttpTrigger(
+                name = "req",
+                methods = {HttpMethod.POST},
+                authLevel = AuthorizationLevel.ANONYMOUS,
+                route = "addproduct")
+                HttpRequestMessage<Optional<String>> request,
+            @SQLOutput(
+                commandText = "Products",
+                connectionStringSetting = "SqlConnectionString")
+                OutputBinding<Product> product) throws JsonParseException, JsonMappingException, IOException {
```
Comment (Charles-Gagnon) [inline samples/samples-java/src/main/java/com/function/AddProduct.java:32]: Out of curiosity where did you get this list of exceptions? I have to imagine that isn't all of them - for example if there's a SqlException in the function itself what kind of exception is thrown in the worker?

Not a huge deal so we can leave it like this for now - just want to know what to expect here.

File: samples/samples-java/src/main/java/com/function/AddProduct.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package com.function;
```
Comment (Charles-Gagnon) [inline samples/samples-java/src/main/java/com/function/AddProduct.java:1]: Copyright for these

Can you create a follow up task to investigate linting options for Java?

**Review by Charles-Gagnon — COMMENTED** [review]
Update the README too

File: samples/samples-java/src/main/java/com/function/AddProduct.java (hunk 3 above)
Comment (lucyzhang929) [reply samples/samples-java/src/main/java/com/function/AddProduct.java:32]: The exceptions come from mapper.readValue() in line 36. If I don't add the throws, there are build errors. Ex: `Unhandled exception type JsonProcessingException`

File: samples/samples-java/src/main/java/com/function/AddProduct.java (hunk 4 above)
Comment (lucyzhang929) [reply samples/samples-java/src/main/java/com/function/AddProduct.java:7]: Tracking here: https://github.com/Azure/azure-functions-sql-extension/issues/462

File: README.md [hunk 5]
```diff
@@ -502,6 +529,148 @@ Note: This tutorial requires that a SQL database is setup as shown in [Create a
...
+- Navigate back to your HttpTriggerJava file.
+- Open the local.settings.json file, and in the brackets for "Values," verify there is a 'SqlConnectionString.' If not, add it.
+- Hit 'F5' to run your code. This will start up the Functions Host with a local HTTP Trigger and SQL Input Binding.
+- Click the link that appears in your terminal.
+- You should see your database output in the browser window.
+- Congratulations! You have successfully created your first SQL input binding! Checkout [Input Binding](#Input-Binding) for more information on how to use it and explore on your own!
+
+#### Output Binding Tutorial
+
+Note: This tutorial requires that a SQL database is setup as shown in [Create a SQL Server](#Create-a-SQL-Server), and that you have the 'Employee.cs' class from the [Input Binding Tutorial](#Input-Binding-Tutorial).
```
Comment (Charles-Gagnon) [inline README.md:639]: `have the 'Employee.java' file`

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**